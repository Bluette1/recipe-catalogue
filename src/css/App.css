@import url('https://fonts.googleapis.com/css2?family=Mountains+of+Christmas&display=swap');
@import url('https://fonts.googleapis.com/css2?family=Elsie&family=Mountains+of+Christmas&display=swap');
@import url('https://fonts.googleapis.com/css2?family=Elsie&family=Milonga&family=Mountains+of+Christmas&display=swap');

ul {
  display: flex;
  list-style: none;
}

.mealslist-heading li,
.mealslist-heading select {
  padding-right: 10px;
  margin-right: 50px;
  width: 59px;
  height: 16px;
  font-family: 'Milonga', cursive;
  font-size: 13px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: normal;
  letter-spacing: 1.9px;
  color: #121212;
  cursor: pointer;
}

.mealslist-heading select {
  width: 140px;
  padding-left: 3px;
  font-weight: bold;
  background-color: #d57d1f;
}

.mealslist-heading li:first-child {
  width: 250px;
  height: 37px;
  font-family: 'Mountains of Christmas', cursive;
  font-size: 30px;
  font-weight: bold;
  font-stretch: normal;
  font-style: normal;
  line-height: normal;
  letter-spacing: normal;
  color: #008000;
  padding-top: 0;
}

.header {
  margin-top: 5%;
  display: flex;
  justify-content: space-between;
  padding-right: 100px;
  padding-left: 60px;
  border-bottom: 1px solid #e8e8e8;
  padding-bottom: 18px;
  background-color: #d57d1f;
}

.content {
  background-color: #000;
  margin-top: -35px;
}

button {
  width: 160px;
  font-family: 'Elsie', cursive;
  font-size: 13px;
  font-weight: 300;
  font-stretch: normal;
  font-style: normal;
  line-height: normal;
  letter-spacing: 0.5px;
  text-align: center;
  color: #fff;
  height: 33px;
  margin-left: -80px;
  margin-right: 190px;
  padding: 7px 19px 8px 22px;
  border-radius: 3px;
  background-color: #2d2013;
  border: #2d2013;
  cursor: pointer;
}
