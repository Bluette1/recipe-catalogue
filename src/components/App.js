import './App.css';
import React from 'react';
import MealsList from '../containers/MealsList';
import CategoryFilter from './CategoryFilter';

export default function App() {
  return (
    <div className="content">
      <div className="header">
        <ul className="mealslist-heading">
          <li>
            Recipe catalogue
          </li>
          <li>Meals</li>
          <CategoryFilter />
        </ul>
        <i className="fas fa-user-circle fa-3x" aria-hidden="true" />
      </div>
      <div>
        <MealsList />
      </div>
    </div>
  );
}
