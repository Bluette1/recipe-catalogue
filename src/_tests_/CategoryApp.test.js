import React from 'react';
import {
  render, screen, cleanup, fireEvent, waitFor,
} from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import axios from 'axios';
import App from '../components/CategoryApp';
import configureStore, { history } from '../store';

configure({ adapter: new Adapter() });

jest.mock('axios');

afterEach(cleanup);

const store = configureStore();

const AppWithStore = () => (
  <Provider store={store}>
    <React.StrictMode>
      <ConnectedRouter history={history}><App /></ConnectedRouter>
    </React.StrictMode>
  </Provider>
);
test('renders CategoryApp', async () => {
  axios.get.mockImplementation(() => Promise.resolve({ data: { meals: [] } }));

  await act(async () => {
    render(<AppWithStore />);
  });
  await waitFor(() => {
    const catalogueTitle = screen.getByText(/Recipe Catalogue/i);
    expect(catalogueTitle).toBeInTheDocument();
    const filter = screen.getByText(/search by/i);
    expect(filter).toBeInTheDocument();
  });
});

it('renders CategoryApp with home button component', async () => {
  axios.get.mockImplementation(() => Promise.resolve({ data: { meals: [] } }));

  const div = document.createElement('div');
  let rendered;
  await act(async () => {
    rendered = render(<AppWithStore />, div);
  });

  await waitFor(() => {
    const homeButton = rendered.getByText('Meals');
    fireEvent.click(homeButton);
    expect(homeButton).toBeDefined();
    expect(rendered).toMatchSnapshot();
  });
});

it('renders CategoryApp with expected component - test with enzyme', async () => {
  axios.get.mockImplementation(() => Promise.resolve({ data: { meals: [] } }));

  let app;
  await act(async () => {
    app = mount(<AppWithStore />);
  });
  await waitFor(() => {
    expect(app.find('button').text()).toBe('Meals');
    const link = app.find('a').get(0);
    expect(link.props.href).toBe('/');
    expect(app.find('.header')).toBeDefined();
    expect(app).toMatchSnapshot();
  });
});
