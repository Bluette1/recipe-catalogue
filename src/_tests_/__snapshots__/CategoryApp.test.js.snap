// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CategoryApp with expected component - test with enzyme 1`] = `ReactWrapper {}`;

exports[`renders CategoryApp with home button component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="content"
      >
        <ul
          class="mealslist-heading"
        >
          <li>
            Recipe Catalogue
          </li>
          <li>
            <a
              href="/"
            >
              <button
                type="button"
              >
                Meals
              </button>
            </a>
          </li>
          <select
            class="category-filter"
          >
            <option>
              SEARCH BY
            </option>
            <option
              value="All"
            >
              All
            </option>
            <option
              value="A"
            >
              A
            </option>
            <option
              value="B"
            >
              B
            </option>
            <option
              value="C"
            >
              C
            </option>
            <option
              value="D"
            >
              D
            </option>
            <option
              value="E"
            >
              E
            </option>
            <option
              value="F"
            >
              F
            </option>
            <option
              value="G"
            >
              G
            </option>
            <option
              value="H"
            >
              H
            </option>
            <option
              value="I"
            >
              I
            </option>
            <option
              value="J"
            >
              J
            </option>
            <option
              value="L"
            >
              L
            </option>
            <option
              value="M"
            >
              M
            </option>
            <option
              value="N"
            >
              N
            </option>
            <option
              value="O"
            >
              O
            </option>
            <option
              value="P"
            >
              P
            </option>
            <option
              value="R"
            >
              R
            </option>
            <option
              value="S"
            >
              S
            </option>
            <option
              value="T"
            >
              T
            </option>
            <option
              value="U"
            >
              U
            </option>
            <option
              value="V"
            >
              V
            </option>
            <option
              value="W"
            >
              W
            </option>
            <option
              value="X"
            >
              X
            </option>
            <option
              value="Y"
            >
              Y
            </option>
            <option
              value="Z"
            >
              Z
            </option>
          </select>
        </ul>
        <div>
          <div />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="content"
    >
      <ul
        class="mealslist-heading"
      >
        <li>
          Recipe Catalogue
        </li>
        <li>
          <a
            href="/"
          >
            <button
              type="button"
            >
              Meals
            </button>
          </a>
        </li>
        <select
          class="category-filter"
        >
          <option>
            SEARCH BY
          </option>
          <option
            value="All"
          >
            All
          </option>
          <option
            value="A"
          >
            A
          </option>
          <option
            value="B"
          >
            B
          </option>
          <option
            value="C"
          >
            C
          </option>
          <option
            value="D"
          >
            D
          </option>
          <option
            value="E"
          >
            E
          </option>
          <option
            value="F"
          >
            F
          </option>
          <option
            value="G"
          >
            G
          </option>
          <option
            value="H"
          >
            H
          </option>
          <option
            value="I"
          >
            I
          </option>
          <option
            value="J"
          >
            J
          </option>
          <option
            value="L"
          >
            L
          </option>
          <option
            value="M"
          >
            M
          </option>
          <option
            value="N"
          >
            N
          </option>
          <option
            value="O"
          >
            O
          </option>
          <option
            value="P"
          >
            P
          </option>
          <option
            value="R"
          >
            R
          </option>
          <option
            value="S"
          >
            S
          </option>
          <option
            value="T"
          >
            T
          </option>
          <option
            value="U"
          >
            U
          </option>
          <option
            value="V"
          >
            V
          </option>
          <option
            value="W"
          >
            W
          </option>
          <option
            value="X"
          >
            X
          </option>
          <option
            value="Y"
          >
            Y
          </option>
          <option
            value="Z"
          >
            Z
          </option>
        </select>
      </ul>
      <div>
        <div />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
